src/interop/react.ts

```diff
@@ -0,0 +1,94 @@
+import type { RimmelComponent, Inputs } from '../types/internal';
+import { BehaviorSubject } from 'rxjs';
+import { Rimmel_Bind_Subtree } from '../lifecycle/data-binding';
+import * as React from 'react';
+
+/**
+ * Wraps a Rimmel component for use in React applications.
+ * 
+ * The wrapped component:
+ * - Converts React props to BehaviorSubject observables
+ * - Maintains Rimmel's reactive model without re-rendering
+ * - Properly cleans up subscriptions on unmount
+ * 
+ * @param rimmelComponent - The Rimmel component function to wrap
+ * @returns A React component that can be used in JSX
+ * 
+ * @example
+ * ```tsx
+ * const ClickCounter = ({ externalCounter }) => {
+ *   const count = new BehaviorSubject(0).pipe(scan(x => x + 1));
+ *   return rml`
+ *     <button onclick="${count}">click</button>
+ *     Count: <output>${count}</output>
+ *     External counter: <output>${externalCounter}</output>
+ *   `;
+ * };
+ * 
+ * const ReactCounter = WrapForReact(ClickCounter);
+ * 
+ * // In React:
+ * <ReactCounter externalCounter={reactState} />
+ * ```
+ */
+export function WrapForReact<P extends Record<string, any> = Record<string, any>>(
+  rimmelComponent: RimmelComponent
+) {
+  return React.forwardRef<HTMLDivElement, P>((props, ref) => {
+    const containerRef = React.useRef<HTMLDivElement>(null);
+    const subjectsRef = React.useRef(new Map());
+    const isMountedRef = React.useRef(false);
+
+    // Merge external ref with internal ref
+    React.useImperativeHandle(ref, () => containerRef.current!);
+
+    // Initialize or update BehaviorSubjects when props change
+    React.useEffect(() => {
+      const subjects = subjectsRef.current;
+
+      // Create or update subjects for each prop
+      Object.keys(props).forEach(key => {
+        const propValue = props[key];
+        
+        if (subjects.has(key)) {
+          // Update existing subject
+          subjects.get(key)!.next(propValue);
+        } else {
+          // Create new subject
+          subjects.set(key, new BehaviorSubject(propValue));
```

Comment:
A component might actually not want to subscribe to every possible prop. How about only creating Subjects/Behaviors when an actual call to .subscribe() is made?

So the check you did above, on line 55, may still hold and would fork fine, with the added benefit that we would not make unnecessary calls to `.next()` on any stream if here's no subscriber...